osquery/sql/sqlite_filesystem.cpp

```diff
@@ -0,0 +1,166 @@
+        result = temp_string;
+        if (shortest) {
+          break;
+        }
+      }
+    }
+    temp_string += path[pos];
+  }
+  if (result.length() == 0 || !shortest) {
+    boost::filesystem::path test_path = temp_string;
+    auto status = boost::filesystem::status(test_path);
+    if (boost::filesystem::exists(status) &&
+        !boost::filesystem::is_directory(status)) {
+      result = temp_string;
+    }
+  }
+  return result;
+}
+
+static boost::optional<std::string> findExistingProgramPathFromCommandSqlArgs(
+    int argc, sqlite3_value** argv, bool shortest) {
+  if (argc == 0) {
+    return boost::none;
+  }
+  // NULLs are not allowed
+  for (int i = 0; i < argc; i++) {
+    if (SQLITE_NULL == sqlite3_value_type(argv[i])) {
+      return boost::none;
+    }
+  }
+  const char* path = (const char*)sqlite3_value_text(argv[0]);
+  bool allow_quoting = false;
+  if (argc > 1) {
+    allow_quoting = sqlite3_value_int(argv[1]) != 0 ? true : false;
+  }
+#ifdef WIN32
+  char escape_symbol = '^';
+#else
+  char escape_symbol = '\\';
+#endif
+  if (argc > 2) {
+    const char* escape_symbol_string = (const char*)sqlite3_value_text(argv[2]);
+    if (escape_symbol_string == NULL ||
+        std::strlen(escape_symbol_string) != 1) {
+      return boost::none;
+    }
+    escape_symbol = escape_symbol_string[0];
+  }
+  return findExistingProgramPathFromCommand(path,
+                                            escape_symbol,
+                                            allow_quoting,
+                                            shortest);
+}
+  
+static void findFilePathInLaunchCommand(sqlite3_context* context,
+                                          int argc,
+                                          sqlite3_value** argv) {
+  auto result = findExistingProgramPathFromCommandSqlArgs(argc, argv, true);
+  if (result) {
+    std::string string = *result;
+    sqlite3_result_text(context,
+                        string.c_str(),
+                        static_cast<int>(string.size()),
+                        SQLITE_TRANSIENT);
+  } else {
+    sqlite3_result_error(
+        context, "Invalid inputs to find_binary_path_from_cmd", -1);
+  }
+}
+
+static void isPathDeterministic(sqlite3_context* context,
+                                int argc,
+                                sqlite3_value** argv) {
+  auto shortest = findExistingProgramPathFromCommandSqlArgs(argc, argv, true);
+  if (shortest) {
+    const char* path = (const char*)sqlite3_value_text(argv[0]);
+    if (shortest->length() == 0 || shortest->length() == strlen(path)) {
+      // There are 2 cases:
+      // 1 - empty string, all parts of path are invalid,
+      // so path is deterministic
+      // 2 - short == full, then there is only 1 valid path
+      sqlite3_result_int(context, 1);
+      return;
+    } else {
+      auto longest =
+          findExistingProgramPathFromCommandSqlArgs(argc, argv, false);
+      if (longest) {
```

Comment:
Same here, prefer the explicit check `.is_initialized()`

Reply:
is_initialized is deprecated method, if (optional) is one of the main features of optionals and preffred way to do it in boost